lib/autocomplete-manager.coffee

```diff
@@ -188,66 +116,46 @@ class AutocompleteManager
 
     # No suggestions? Cancel autocompletion.
     return unless suggestions.length
+    @suggestionList?.changeItems(suggestions)
+    @showSuggestionList()
+
+  providersForScopes: (scopes) =>
+    return [] unless scopes?
+    return [] unless @scopes
+    providers = []
+    for scope in scopes
+      if @scopes[scope]?
+        providers = _.union(providers, @scopes[scope])
+    providers.push(@fuzzyProvider) unless _.size(providers) > 0
+    providers
 
-    unless @overlayDecoration?
-      marker = @editor.getLastCursor()?.getMarker()
-      @overlayDecoration = @editor?.decorateMarker(marker, { type: 'overlay', item: this })
-
-    # Now we're ready - display the suggestions
-    @changeItems suggestions
-
-    @setActive()
-
-  changeItems: (items) ->
-    @items = items
-    @emitter.emit 'did-change-items', items
-
-  onDidChangeItems: (cb) ->
-    @emitter.on 'did-change-items', cb
-
-  # Private: Focuses the hidden input, starts listening to keyboard events
-  setActive: ->
-    @addKeyboardInteraction()
-    @active = true
+  # Private: Gets called when the user successfully confirms a suggestion
+  #
+  # match - An {Object} representing the confirmed suggestion
+  confirm: (match) =>
+    return unless @editorHasFocus()
+    return unless match?.provider?
```

Comment:
You'd be surprised how often it occurs during development of a new provider. This will likely change soon, as we're going to switch to a non-`Provider` (class) based approach.